pkg/cluster/agent_customization.go

```diff
@@ -0,0 +1,86 @@
+package cluster
+
+import (
+	"encoding/json"
+
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/settings"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// GetClusterAgentTolerations returns additional tolerations for the cluster agent if they have been user defined. If
+// not, nil is returned.
+func GetClusterAgentTolerations(cluster *v3.Cluster) []corev1.Toleration {
+	if cluster.Spec.ClusterAgentDeploymentCustomization != nil &&
+		cluster.Spec.ClusterAgentDeploymentCustomization.AppendTolerations != nil {
```

Comment:
Technically we don't need check for `cluster.Spec.ClusterAgentDeploymentCustomization.AppendTolerations != nil` as even if we don't do it we will return `nil` regardless. But I agree it's safer this way and follows the logic of other `GetClusterAgent...` functions where there is a default to return.
Nothing to address here, just a comment.